### ISSUE TITLE
The use of BlurPool

### ISSUE THREAD

**Issue by WW2401** [ricky40403/BlurPool#1]
Great work. Thanks for your sharing.
When I use BlurPool, it only needs to change caffe.proto and replace the original base_conv_layer.cpp? And you mentioned that Caffe uses zero paddings instead of other types, can you tell me the deficiency of using zero paddings?

**Comment by ricky40403** [comment]
1. Yes, and should freeze the blur convolution weight when training, or it will change during backpropagation.

2. The only drawback should be that it can not increase the resolution. And the padding from default is set to 'reflect' in origin PyTorch repository.

**Comment by WW2401** [comment]
> 1. Yes, and should freeze the blur convolution weight when training, or it will change during backpropagation.
> 2. The only drawback should be that it can not increase the resolution. And the padding from default is set to 'reflect' in origin PyTorch repository.

thanks for your reply. Have you trained the net with blurpool? How about the performance? speed and accuracy

**Comment by ricky40403** [comment]
> > 1. Yes, and should freeze the blur convolution weight when training, or it will change during backpropagation.
> > 2. The only drawback should be that it can not increase the resolution. And the padding from default is set to 'reflect' in origin PyTorch repository.
> 
> thanks for your reply. Have you trained the net with blurpool? How about the performance? speed and accuracy

Yes, I had trained on my custom dataset, and the performance  is indeed better. The accuracy and the shifting problem become better on the inference.
But I also trained with sparse, so I can not tell how the speed is in the general model.

By the way, I was training with segmentation, not classification.
